pyomo/contrib/appsi/base.py

```diff
@@ -767,8 +771,8 @@ def set_objective(self, obj: _GeneralObjectiveData):
             self._set_objective(obj)
 
     def add_block(self, block):
-        self.add_variables([var for var in block.component_data_objects(Var, descend_into=True, sort=False)])
-        self.add_params([p for p in block.component_data_objects(Param, descend_into=True, sort=False)])
+        self.add_variables(ComponentSet(var for var in block.component_data_objects(Var, descend_into=True, sort=False)))
+        self.add_params(ComponentSet(p for p in block.component_data_objects(Param, descend_into=True, sort=False)))
```

Comment:
Does it matter that this is could be a nondeterministic order in Python 3.6?

Reply:
Yes. This does matter. Good catch. I will used the OrderedSet instead.